bin/smack.py

```diff
@@ -141,6 +144,9 @@ def arguments():
   verifier_group.add_argument('--loop-limit', metavar='N', default='1', type=int,
     help='upper bound on minimum loop iterations [default: %(default)s]')
 
+  verifier_group.add_argument('--context-switches', metavar='k', default='1', type=int,
```

Comment:
I think there are many possible hypotheses to be made here, depending, e.g., on what is Corral’s default schedule of threads (is `main` first, or last?), e.g., on whether `main` must complete its execution (and thus have joined both threads), e.g., on whether a switch is counted when a thread completes, etc. My hunch, mainly driven by how Corral describes its own flag, is that the parameter is the number of execution contexts *per thread*. So for instance, supposing that `main` must complete its execution (and also depending on other factors, like your implementation of `pthreads`) the `main` thread must be given two contexts — the first to create the other threads, and the second to join them. But there are many moving parts here, and I have not reviewed thoroughly your `pthreads` implementation, nor the inner workings of Corral.

Above all though, it cannot be right to call this parameter “context switches” when the default value of `1` is used for sequential programs — which do not do context switches.

So perhaps it’s best that we don’t make a firm call on the precise meaning of this parameter, at least not just yet. The safest thing would be to remain vague, but not too vague. Maybe `--context-bound` would be a good choice?